tests/test_tools.py

```diff
@@ -144,3 +148,18 @@ def test_unbound_tuple():
     assert ub.next() == 2
     ub.iter()
     assert ub.next() == 1
+
+
+def test_temp_order():
```

Comment:
This kinda belongs to test_dse

If not test_dse, definitely not here. The tools are those inside devito/tools